src/Symfony/Component/DependencyInjection/Compiler/ResolveAutowireInlineAttributesPass.php

```diff
@@ -52,18 +52,13 @@ protected function processValue(mixed $value, bool $isRoot = false): mixed
             }
         }
 
-        $dummy = $value;
-        while (null === $dummy->getClass() && $dummy instanceof ChildDefinition) {
```

Comment:
this is dead code: definitions with no class are already skipped before (and that's legit for child definitions too)